src/doc/rustdoc/src/write-documentation/re-exports.md

```diff
@@ -0,0 +1,128 @@
+    pub struct Foo;
+}
+pub mod sub_module2 {
+    pub struct AnotherFoo;
+}
+```
+
+Users can import them like this:
+
+```rust,ignore (inline)
+use lib::sub_module1::Foo;
+use lib::sub_module2::AnotherFoo;
+```
+
+But what if you want the types to be available directly at the crate root or if we don't want the
+modules to be visible for users? That's where re-exports come in:
+
+```rust,ignore (inline)
+// `sub_module1` and `sub_module2` are not visible outside.
+mod sub_module1 {
+    pub struct Foo;
+}
+mod sub_module2 {
+    pub struct AnotherFoo;
+}
+
+// We re-export both types:
+pub use crate::sub_module1::Foo;
+pub use crate::sub_module2::AnotherFoo;
+```
+
+And now users will be able to do:
+
+```rust,ignore (inline)
+use lib::{Foo, AnotherFoo};
+```
+
+And since both `sub_module1` and `sub_module2` are private, users won't be able to import them.
+
+Now what's interesting is that the generated documentation for this crate will show both `Foo` and
+`AnotherFoo` directly at the crate root, meaning they have been inlined. There are a few rules to
+know whether or not a re-exported item will be inlined.
+
+## Inlining rules
+
+If a public item comes from a private module, it will be inlined:
+
+```rust,ignore (inline)
+mod private_module {
+    pub struct Public;
+}
+
+pub mod public_mod {
+    // `Public` will inlined here since `private_module` is private.
+    pub use super::private_module::Public;
+}
+
+// `Public` will not be inlined here since `public_mod` is public.
+pub use self::public_mod::Public;
+```
+
+Likewise, if an item has `#[doc(hidden)]` or inherits it (from any of its parents), it
+will be inlined:
+
+```rust,ignore (inline)
+#[doc(hidden)]
+pub mod public_mod {
+    pub struct Public;
+}
+
+#[doc(hidden)]
+pub struct Hidden;
+
+// `Public` be inlined since its parent (`public_mod`) has `#[doc(hidden)]`.
+pub use self::public_mod::Public;
+// `Hidden` be inlined since it has `#[doc(hidden)]`.
+pub use self::Hidden;
+```
+
+The inlining rules are a bit different for glob re-exports (`pub use x::*`) for `#[doc(hidden)]`
+types. If we take the previous example and then re-export like this:
+
+```rust,ignore (inline)
+pub use self::*; // It will not inline the `Hidden` struct.
+pub use self::public_mod::*; // It will inline the `Public` struct.
+```
+
+It only impacts elements that have the `#[doc(hidden)]` attributes. If it only inherits it, then it
+is inlined.
+
+## Inlining with `#[doc(inline)]`
+
+You can use the `#[doc(inline)]` attribute if you want to force an item to be inlined:
+
+```rust,ignore (inline)
+pub mod public_mod {
+    pub struct Public;
+}
+
+#[doc(inline)]
+pub use self::public_mod::Public;
+```
+
+With this code, even though `public_mod::Public` is public and present in the documentation, the
+`Public` type will be present both at the crate root and in the `public_mod` module.
+
+## Attributes
+
+When an item is inlined, its doc comments and most of its attributes will be inlined along with it:
```

Comment:
I think this section is hard to understand. It talks about whether "attributes are inlined" but it's more about attributes whose _effects_ are inlined, or if they are themselves effecting inlining.

@notriddle do you have any ideas on how to better frame this?

Reply:
I just think that the table below is heterogenous, with the "inlined?" column not quite making sense in a consistent way across entries.

Reply:
I replaced "Inlined?" with "Is it inlined alongside its item?".